### DISCUSSION TITLE
[A/L] (2021/22) Combinatoria - Esercizio 23

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#199] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152254142-0a890bb7-651e-4f2b-8b74-d757cf04705e.png)

**Comment by LuginiAndrea** [comment]
Considerando l'ordine come influente nella scelta delle parole, divido per possibili casi in base al numero di lettere nelle parole:
1l - 7l
2l - 5l
3l - 4l
Ora considero quanti modi ho di formare una parola con 0 a, 1 a, 2 a, 3 a in quanto a termine che si ripete:
|        Caso        |          0            |          1            |          2             |          3            |
| --------------- | --------------- | --------------- | ---------------  | --------------- |
| 1 lettera          |           5            |         1           |          0              |           0           |
| 2 lettere          |         5x4          |  5x2 (possibili posizioni di a) | 1 | 0         |
| 3 lettere | 5x4x3 | 5x4x3 (possibili posizioni di a) | 5x3x2/2 | 1 |
| 4 lettere | 5x4x3x2 | 5x4x3x4 | 5x4x4x3/2 | 5x4x3x2/6 |
| 5 lettere | 5! | 5x4x3x2x5 | 5x4x3x5x4/2 | 5x4x5x4x3/6 |
| 6 lettere | 0 | 6! | 5x4x3x2x6x5/2| 5x4x3x6x5x4/6|
| 7 lettere | 0 | 0 | 7! / 2 | 5x4x3x2x7x6x5/6|

S = 1L * 7L + 2L * 5L + 3L * 4L

**Comment by Elia-Belli** [comment]
- Possiamo vedere il problema in questo modo: 

> "In quanti modi puoi riordinare le lettere della parola Anarchia e dividerle in 2 parti in modo che ogni parte abbia almeno una lettera?"

   Adesso è ovvio vedere che si tratti di dividere a metà degli anagrammi: anagrammando Anarchia si ottengono sempre parole di 8 lettere e i modi per dividerle in 2 sono sempre 7.

   Tipizziamo in base agli anagrammi (1 anagramma = 1 tipo), abbiamo $8!/3!$ anagrammi, ogni anagramma puoi essere diviso in 7 modi diversi quindi:
Soluzione = $7(8!/3!)$

**Comment by DavideGalilei** [comment]
Un altro modo di risolvere il problema è quello di aggiungere un trattino `-` come separatore alla parola, e calcolare gli anagrammi della parola contenente il separatore:
Un esempio di anagramma diverrebbe $\text{ACN-AIRHA}$, però ogni parola nella coppia deve essere formata da almeno una lettera, pertanto casi come $\text{-ARANCIAH}$ dove il separatore è presente all'inizio o alla fine della parola sono inaccettabili, e bisogna escluderli.

- Anagrammi totali contenenti il separatore: $$\large{\frac{9!}{3! \cdot 1! \cdot 1! \cdot 1! \cdot 1! \cdot 1! \cdot 1!} = \frac{9!}{3!}}$$
- Anagrammi di parole non valide (trattino all'inizio + trattino alla fine): $$\large{\frac{8!}{3!} + \frac{8!}{3!} = 2 \cdot \frac{8!}{3!}}$$
- Risultato: $$\text{CoppieDiParole} = \text{AnagrammiTotali} - \text{ParoleNonValide} = \large{\frac{9!}{3!} - 2 \cdot \frac{8!}{3!} = 47040}$$
